lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -2816,6 +2816,11 @@ public function addDiscriminatorMapClass($name, $className)
             throw MappingException::invalidClassInDiscriminatorMap($className, $this->name);
         }
 
+        $refl = new ReflectionClass($className);
```

Comment:
Sorry to swoop in so late, but loading ReflectionClass inside ClassMetadataInfo is not allowed from architecture POV. This code doesn't work with static runtime reflection, which will break detached metadata use cases (such as everything with the EntityGenerator).

Instead this check is something you should put either into ClassMetadataFactory if we want this to fail at runtime, or in the SchemaValidator if we want this to fail only in debug/dev mode.